### ISSUE TITLE
Brodcast to specific areas or devices ARE SUPPORTED

### ISSUE THREAD

**Issue by Mightyinks** [greghesp/assistant-relay#224]
Hello, 
I need to point out that broadcast to specific google home is actually supported. In the manual says no and I know it used to be the case but this can now be corrected:

**_Limitations#
By Google's design, a broadcast goes out to all Google Home devices. This cannot be changed._**

Thanksfully it has been supported for a while now and in the assistant relay you can use a command instead of the Broadcast option to send your message to specific ones.
 Like this 
By room >> Command: broadcast to living room, are surprised yet? 
This is choosing an specific room that the google homes are. but in my case some rooms have multiples so for this I use the following

By Device >> Command: Broadcast to living room speaker, this is a TEST.
 or even hubs using their name for example >> Command: Broadcast to Studio HUB, this is a TEST. 
Notice that I use my specific device name so you need to use yours.

**Comment by ryancasler** [comment]
Yes....did you look at the date that GAR was released?  That was before this change from Google.  The new version that is being worked on will have native support for specific broadcasts.  But if you want to speed things along, you're more than welcome to take a look at the help request that Greg has and do a pull request with updates.  Otherwise, be patient.

**Comment by Mightyinks** [comment]
> Yes....did you look at the date that GAR was released? That was before this change from Google. The new version that is being worked on will have native support for specific broadcasts. But if you want to speed things along, you're more than welcome to take a look at the help request that Greg has and do a pull request with updates. Otherwise, be patient.

I was trying to help those new ones that might not know that they can achieve this now without the need of any updates or anything. I don't need to be patient because there's nothing I need, it currently does everything perfectly 👍

**Comment by Mightyinks** [comment]
Try it for an specific room like example > Broadcast to Kitchen hello there
my friend.
I'm using it like that and it works so I created custom pistons in Webcore
and it makes announcements directly to those rooms and I plan to create
location triggers using NFC tags to specify in what room I am so it send me
notifications like calendar events, when the mail arrives and more only to
the room that I am currently at. The fact that the specific room broadcast
works is exciting for me.

**Comment by rezzalex** [comment]
Don't work for me, neither with the room name.
Are we on the same version ?


Le ven. 22 janv. 2021 à 18:29, Mightyinks <notifications@github.com> a
écrit :

> Try it for an specific room like example > Broadcast to Kitchen hello there
> my friend.
> I'm using it like that and it works so I created custom pistons in Webcore
> and it makes announcements directly to those rooms and I plan to create
> location triggers using NFC tags to specify in what room I am so it send me
> notifications like calendar events, when the mail arrives and more only to
> the room that I am currently at. The fact that the specific room broadcast
> works is exciting for me.
>

**Comment by Mightyinks** [comment]
That’s super wierd. I will probably will have to make a video and upload it
to show it working and my process and versions.

**Comment by ryancasler** [comment]
Has nothing to do with GAR.  This is a function of Google Assistant.  What region/language are you located in?  Room/device specific broadcasts have not rolled out to all regions/languages as of yet.

**Comment by Mightyinks** [comment]
Oh ok! My system firmware version is 229149 and cast version is 1.50.229149
in one of the devices I use. Im located in the USA and using it in English.
I also have a Lenovo Hub that it works even showing the broadcast in text
which is has a system version OIMI.200528.001 and cast version 1.52.239764.

**Comment by ryancasler** [comment]
> Don't work for me, neither with the room name. It indeed broadcast the message to ALL available speakers saying "here is the message from "$USER", to "$room" or "$SPEAKER", "$Rest of the message" ". Are we on the same version ?

I just reread your message.  You are passing the correct values to AR correct?  So, if I want to broadcast a message to the device named "Office display" I have to send the command, with broadcast turned **OFF** (because using broadcast will force it to all your speakers).  So, in the sandbox, it looks like this:
![image](https://user-images.githubusercontent.com/30270489/105663389-17159200-5ea0-11eb-8905-0816fb033edd.png)


And if you're looking at the body of the request to send, the json should only contain your username and the command with all of the details in the command field.

**Comment by Mightyinks** [comment]
You are exactly right! I think I failed to point that out sorry lol

**Comment by rezzalex** [comment]
> version

On my side, all Google devices registered in France.

a Lenovo Smart Display with those firmwares :
   - system : 33.27.44.346118233
   - cast :  1.52.239754

A Google Home mini : 
  - system : 229149
   - cast :  1.50.229149

**Comment by rezzalex** [comment]
> > Don't work for me, neither with the room name. It indeed broadcast the message to ALL available speakers saying "here is the message from "$USER", to "$room" or "$SPEAKER", "$Rest of the message" ". Are we on the same version ?
> 
> I just reread your message. You are passing the correct values to AR correct? So, if I want to broadcast a message to the device named "Office display" I have to send the command, with broadcast turned **OFF** (because using broadcast will force it to all your speakers). So, in the sandbox, it looks like this:
> ![image](https://user-images.githubusercontent.com/30270489/105663389-17159200-5ea0-11eb-8905-0816fb033edd.png)
> 
> And if you're looking at the body of the request to send, the json should only contain your username and the command with all of the details in the command field.
> 
> Also, for some reason, mine has problems when the first word after the device name is "This". So, if i try to say "This is a test, I have to put this twice. That's because it think you are saying broadcast this message and not that "this" is part of the message. Stupid, I know, but how often does your message start with the word "This".

Almost what I did initially...., except I wrote "Broadcast to" instead of " Broadcast on", but... same result...

**Comment by rezzalex** [comment]
Finally using v3.3.2b, and by replacing "Broadcast to" by ma configured language (french), it worked.

can be closed concerning me.
thanks a lot
